tools/upload.py

```diff
@@ -45,25 +45,30 @@
     elif thisarg == 'write_flash':
         write_addr = sys.argv.pop(0)
         binary = sys.argv.pop(0)
-    elif len(thisarg):
+    elif thisarg:
         cmdline = cmdline + [thisarg]
 
 cmdline = cmdline + ['write_flash']
-if len(write_option):
+if write_option:
     cmdline = cmdline + [write_option]
 cmdline = cmdline + ['--flash_size', 'detect']
 cmdline = cmdline + [write_addr, binary]
 
 erase_file = ''
-if len(erase_addr):
+if erase_addr:
     # Generate temporary empty (0xff) file
     eraser = tempfile.mkstemp()
     erase_file = eraser[1]
-    os.write(eraser[0], bytearray([255] * int(erase_len, 0)))
+    os.write(eraser[0], bytearray([0xff] * int(erase_len, 0)))
     os.close(eraser[0])
-    cmdline = cmdline + [ erase_addr, erase_file ]
+    cmdline = cmdline + [erase_addr, erase_file]
 
-esptool.main(cmdline)
-
-if len(erase_file):
-    os.remove(erase_file)
+try:
+    esptool.main(cmdline)
+except esptool.FatalError as e:
+    sys.stderr.write('\nA fatal esptool.py error occurred: %s' % e)
+finally:
+    if len(erase_file):
```

Comment:
Should this just be `if erase_file:` to mirror the other changes you've done to Python-ify the script?